provider/aws: Add Lightsail Instance

catsby opened the pull request:
Support for [AWS Lightsail](https://lightsail.aws.amazon.com/ls/docs/all)

Includes:

- [x] Docs for `aws_lightsail_instance`
- [x] Tests for `aws_lightsail_instance`
- [x] `aws_lightsail_instance` resource

catsby commented:
We're limited on `key_pair` right now, we'll need a separate resource for that

stack72 commented on builtin/providers/aws/resource_aws_lightsail_instance.go:
```diff
@@ -0,0 +1,264 @@
+		req.KeyPairName = aws.String(v.(string))
+	}
+	// // encoding required?
+	// if v, ok := d.GetOk("user_data"); ok {
```
This intentionally missing? :)

stack72 commented on builtin/providers/aws/resource_aws_lightsail_instance.go:
```diff
@@ -0,0 +1,264 @@
+		log.Printf("[ERR] Error waiting for instance (%s) to become ready: %s", d.Id(), err)
+	}
+
+	// return resourceAwsLightsailInstanceUpdate(d, meta)
```
worth removing this old comment?

catsby replied:
whoops :)

catsby commented:
@stack72 thanks for the review, I've updated it!

> 1 other thing, is it worth adding a few more tests? This would allow us to check for the _disappears

I added `TestAccAWSLightsailInstance_disapear` for this 

> import

The `_basic` test has the `IDRefreshName` thing, any more than that?

> maybe an update

Sadly everything is `ForceNew` at the moment, the API doesn't offer much for updates at all. The resource lacks an `Update` method

stack72 commented:
LGTM! thanks for the changes :)
